lib/tasks/trn_generator.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class TRNGenerator
+  class << self
+    def next
+      available.pop || (raise "TRN available list exhausted")
+    end
+
+  private
+
+    ALL_TRNS = ("0011111".."9999999").to_a.freeze unless defined?(ALL_TRNS)
+
+    def available
+      @available ||= (ALL_TRNS - taken).shuffle
+    end
+
+    def taken
+      @taken ||= TeacherProfile.where.not(trn: nil).pluck(:trn)
```

Comment:
if you memoize this it means the query result is cached. this means when you call `next`, `available` is not updated therefore you can generate duplicate trns. is that desired?

Reply:
Check again. You're missing the `available.pop` which is used by next